packages/venia-concept/src/components/CategoryList/categoryList.js

```diff
@@ -0,0 +1,105 @@
+import { Component, createElement } from 'react';
+import { string, number, shape } from 'prop-types';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import classify from 'src/classify';
+import defaultClasses from './categoryList.css';
+
+// TODO: get only active categories from graphql when it is ready
+const categoryListQuery = gql`
+    query category($id: Int!) {
+        category(id: $id) {
+            children {
+                name
+                url_key
+                url_path
+                image
+            }
+        }
+    }
+`;
+
+// TODO: get baseUrl from graphql when it is ready
+const baseUrl = 'https://magento-venia.local.pwadev:8000';
+
+// TODO: get mediaUrl from graphql when it is ready
+const mediaUrl = 'https://magento-venia.local.pwadev:8000/pub/media';
+
+// TODO: get categoryUrlSuffix from graphql when it is ready
+const categoryUrlSuffix = '.html';
+
+class CategoryList extends Component {
+    static propTypes = {
+        id: number,
+        title: string,
+        classes: shape({
+            root: string,
+            header: string,
+            content: string,
+            title: string,
+            item: string,
+            imageWrapper: string,
+            image: string,
+            name: string
+        })
+    };
+
+    static defaultProps = {
+        id: 2
+    };
+
+    render() {
+        const { id, classes } = this.props;
+
+        return (
+            <div className={classes.root}>
+                {this.props.title && (
+                    <div className={classes.header}>
+                        <h2 className={classes.title}>
+                            <span>{this.props.title}</span>
+                        </h2>
+                    </div>
+                )}
+                <Query query={categoryListQuery} variables={{ id }}>
+                    {({ loading, error, data }) => {
+                        if (error) return <div>Data Fetch Error</div>;
+                        if (loading) return <div>Fetching Data</div>;
+                        if (data.category.children == '')
+                            return <div>Here are not any child categories</div>;
+
+                        return (
+                            <div className={classes.content}>
+                                {data.category.children.map((item, index) => (
```

Comment:
How would you feel about implementing the inner function of `.map` here as another Component instead of inline JSX? It's not mandatory, but it's something to think about.

Reply:
Yep, awesome idea.

Reply:
I know we've merged this already, but yes—in general we try to avoid performing work inside JSX expressions. Eventually, I'd like to cover that with a lint rule. Certainly not a dealbreaker, though, so thanks for your understanding. 😄